request_options.go

```diff
@@ -70,3 +70,12 @@ func WithRequestLogger(logger *zap.Logger) RequestOption {
 		return nil
 	}
 }
+
+// WithHooks sets a hook on the request that FrankenPHP will call back
+func WithHooks(hooks *FrankenPHPHooks) RequestOption {
```

Comment:
Why not just `WithRequestFilter` then? No need for an extra layer of indirection IMHO.